### PR TITLE
Implement a basic migrator that can read SQL files from a directory

### PR DESCRIPTION
**Pull request by reagent**
Based off the approach I took for a [Postgres migrator in TypeScript](https://github.com/reagent/migrator#readme), this is my attempt at building a library in Rust to use in other projects that require a Postgres database.

## Usage

Make sure you have a database:

```
createdb migrations_development && \
  echo "DATABASE_URL=postgres://username@localhost/migrations_development" > .env
```

Point the migrator at your migrations directory and Postgres connection:

```rust
use std::path::Path;

use dotenv::dotenv;
use migrator::Migrator;
use postgres::{Client, NoTls};

fn main() {
    dotenv().ok();

    let migrations_path = Path::new(file!()).parent().unwrap();
    let migrations_path = migrations_path.join("db").join("migrations");

    let database_url = dotenv::var("DATABASE_URL").unwrap();

    let mut conn = Client::connect(&database_url, NoTls).unwrap(); // This can fail when db conn fails

    let migrator = Migrator::new(migrations_path);
    migrator.migrate(&mut conn);
}
```

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,164 @@
...
+    pub fn migrate(&self, conn: &mut Client) {
+        let table = MigrationsTable::new("migrations");
+        table.create(conn).unwrap(); // TODO: handle
+
+        for file in self.files() {
+            if !table.was_migration_applied(conn, &file) {
+                table.apply_migration(conn, &file);
+            }
+        }
+    }
```
Comment (reagent) [inline src/lib.rs]: This is the main entry point to find and apply migrations.

File: src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+pub struct Migrator {
+    path: PathBuf,
+}
+
+impl Migrator {
+    pub fn new(path: PathBuf) -> Migrator {
+        Migrator { path }
+    }
+
+    // TODO: return Result
```
Comment (prayash) [inline src/lib.rs]: Pls address this TODO before merge

File: src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+    }
+
+    fn files(&self) -> Vec<MigrationFile> {
+        let mut migration_files: Vec<MigrationFile> = Vec::new();
+
+        if let Ok(files) = fs::read_dir(&self.path) {
+            for file in files {
+                let file = file.unwrap();
+
+                if let Some(file) = MigrationFile::from(file) {
```
Comment (prayash) [inline src/lib.rs]: Why `Some` and not `Ok` here?

File: src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+        transaction.execute(&stmt, &[&file.id]).unwrap();
+        transaction.commit().unwrap();
+    }
+}
+
+impl MigrationFile {
+    fn from(possible: DirEntry) -> Option<MigrationFile> {
+        if let Ok(metadata) = possible.metadata() {
+            if !metadata.is_file() {
+                return None;
```
Comment (prayash) [inline src/lib.rs]: Why `Optional` and not just throw if it's not a file?

**Review by prayash — APPROVED** [review]
LGTM bud but pls address comments 🤙

File: src/lib.rs (hunk 3 above)
Comment (reagent) [reply src/lib.rs]: Because it returns an `Option` generic and not `Result` you dummy

File: src/lib.rs (hunk 4 above)
Comment (reagent) [reply src/lib.rs]: Maybe in VR land you can be sloppy but get that shit outta here 👎🏻

**Pull request closed without merging**